pivot.py

```diff
@@ -0,0 +1,1723 @@
+                    title(self.title)
+                if site.canonical:
+                    meta(name='canonical', href=f'{site.url} {site.canonical}')
+                if site.author:
+                    meta(name='author', content=site.author)
+                #script(src="{{ url_for('static',
+                #   filename='bower_components/jquery/dist/jquery.min.js') }}")
+                #script(src="{{ url_for('static', filename='js/nantic.js') }}")
+                comment('CSS')
+                link(href="/static/output.css", rel="stylesheet")
+                script(src='https://unpkg.com/htmx.org@2.0.0')
+                script(src="https://cdn.tailwindcss.com")
+            main_body = div(id='main', cls='bg-white')
+            #main_body += Header().tag()
+            flash_div = div(id='flash_messages',
+                cls="flex w-full flex-col items-center space-y-4 sm:items-end")
+            flash_div['hx-swap-oob'] = "afterbegin"
+            main_body += flash_div
+            main_body += self.main
+            #main_body += Footer().tag()
+        return html_layout
+
+class Index(Component):
+    'Index'
+    __name__ = 'www.index'
+    _path = None
+
+    database_name = fields.Char('Database Name')
+    table_name = fields.Char('Table Name')
+    table_properties = fields.Char('Table Properties')
+
+    @classmethod
+    def get_url_map(cls):
+        return [
+            Rule('/<string:database_name>/babi/pivot/<string:table_name>/<string:table_properties>')
+        ]
+
+    def render(self):
+        pool = Pool()
+        Layout = pool.get('www.layout')
+        Pivot = pool.get('www.pivot_table')
+        PivotHeader = pool.get('www.pivot_header')
+        BabiTable = pool.get('babi.table')
+        User = pool.get('res.user')
+        Index = pool.get('www.index')
+
+        # Get the table name
+        # If table name starts with "__" we need to remove this part to search the table name
+        table_name = self.table_name
+        if table_name.startswith('__'):
+            table_name = table_name.split('__')[-1]
+
+        babi_tables = BabiTable.search([
+            ('internal_name', '=', table_name)], limit=1)
+        if babi_tables:
+            babi_table = babi_tables[0]
+            table_name = babi_table.name
+
+            users = User.search([('id', '=', Transaction().user)], limit=1)
+            user = users[0] if users else None
+            error_page = False
+            if not user:
+                error_page = True
+            if not error_page and user and babi_table.access_users:
+                if user not in babi_table.access_users:
+                    error_page = True
+            if not error_page and user and babi_table.access_groups:
+                error_page = True
+                for group in user.groups:
+                    if group in babi_table.access_groups:
+                        error_page = False
+                        break
```

Comment:
Access checks should be a method in babi.table that recevied an optional user object (if not provided would just use Transaction().user).